### PR TITLE
fix(058): restore live whiteboard authoring during creation

### LINKED ISSUE #10206: BUG: Whiteboard is not editable in the create whiteboard-callout dialog (regression)
**Describe the bug**
Regression: when creating a whiteboard callout (callout with whiteboard responses), the whiteboard shown inside the create dialog is no longer editable. Up to yesterday (2026-08-25) it was possible to edit the whiteboard content directly in the actual whiteboard within the create-callout dialog; now editing is disabled and the initial whiteboard content can no longer be authored during creation.

**To Reproduce**
1. In a space, start creating a new callout
2. Choose the whiteboard type (callout with whiteboard responses)
3. In the create dialog, try to edit the whiteboard content in the embedded whiteboard
4. Editing is disabled — the whiteboard cannot be modified

Framing:

Image

Responses:

Image

**Expected behavior**
As before, the whiteboard inside the create-callout dialog is editable, so the whiteboard content can be set while creating the callout.

**Additional context**
- Worked up to 2026-08-25; broken as of 2026-08-26 — likely a recent regression.
- Screenshots to follow as a comment.

### PR DESCRIPTION
**Pull request by antst**
## Outcome

Restores the pre-regression whiteboard authoring experience for callout and template creation without sending Yjs or Excalidraw content through GraphQL.

- Selecting Whiteboard performs no mutation.
- Clicking the established large preview or its bottom-right Edit button lazily creates the server-owned live draft and opens the existing editor.
- Restores the original framing UI; no Configure or Ready placeholder is introduced.
- Response defaults keep the established single Edit-button presentation.
- Final submit consumes the draft by ID; cancel deletes it; abandoned drafts expire server-side.
- Templates remain unchanged until Edit is clicked and use the same draft lifecycle when edited.
- Uses a draft-specific query that does not require an attached profile URL.

- Fixes #10206.

- Delivery links:
- Server: https://github.com/alkem-io/server/pull/6405
- Workspace specification: https://github.com/alkem-io/agents-hq/pull/127

## Verification

- Normal signed commit hook passed: lint-staged, TypeScript, ESLint, and full Vitest.
- Full Vitest: 352 files; 2,983 passed; 2 skipped.
- Feature tests prove render and type selection cause zero draft mutation; clicking preview/Edit materializes once.
- Merged current develop into the feature branch; merge hook passed.
- Local browser acceptance remains unclaimed because the available local authentication session expired.

## Summary by CodeRabbit

* **New Features**
  * Added support for creating and editing live Whiteboard drafts while configuring callouts and templates.
  * Drafts are automatically saved with new content and cleaned up when cancelled or no longer needed.
  * Added a dedicated Whiteboard draft editor with loading and navigation states.
  * Improved submission handling to prevent completion while drafts are still processing.

* **Bug Fixes**
  * Prevented dialogs from closing when draft cleanup fails.
  * Added safeguards against duplicate draft creation and incorrect draft deletion.

### CODE REVIEW

**Comment by antst** [comment]
@coderabbitai review

File: schema.graphql [hunk 1]
```diff
@@ -3400,6 +3408,10 @@ type InnovationFlowStateSettings {
   """Ordered widgets shown in the Space sidepanel for this State. May be empty."""
   sidebar: [SidebarWidget!]!
   """
+  Ordered widgets shown in the Space sidepanel for this State. May be empty.
+  """
+  sidebar: [SidebarWidget!]!
+  """
```
[the remark this answers is not in the document]
Comment (antst) [reply schema.graphql]: Fixed in 569792057: removed the duplicate  field from the merged schema;  and the full client gate are green.

File: src/crd/forms/callout/types.ts [hunk 2]
```diff
@@ -94,4 +94,6 @@ export type ContributionDefaults = {
   sourceCalloutId?: string;
   /** Explicit removal; omission preserves an existing default on update. */
   clearWhiteboardContent?: boolean;
+  /** Server-owned live draft used while authoring a default Whiteboard. */
+  whiteboardDraft?: import('@/domain/collaboration/whiteboard/WhiteboardDraft/useWhiteboardDraft').WhiteboardDraftHandle;
```
[the remark this answers is not in the document]
Comment (antst) [reply src/crd/forms/callout/types.ts]: Fixed in 569792057: the CRD layer now owns a local structural  type instead of importing the domain integration type.

File: src/main/crdPages/space/callout/ResponseDefaultsConnector.tsx [hunk 3]
```diff
@@ -115,6 +129,47 @@ export function ResponseDefaultsConnector({
...
+              sourceWhiteboardID: draft.sourceWhiteboardId,
+              sourceCalloutID: draft.sourceCalloutId,
+            });
+            if (!materialized) return;
+            applyDraft({
+              whiteboardDraft: materialized,
+              whiteboardContentAvailable: true,
+              clearWhiteboardContent: false,
+            });
+            setWhiteboardEditorOpen(true);
```
[the remark this answers is not in the document]
Comment (antst) [reply src/main/crdPages/space/callout/ResponseDefaultsConnector.tsx]: Fixed in 569792057: async draft materialization is scoped to the live dialog session, close/cancel invalidates it, and a regression proves a late completion cannot apply values or reopen the editor.

**Comment by antst** [comment]
CodeRabbit triage completed for the remaining review findings.

- Fixed the `handleRef.current` render-time write: synchronization now happens after commit, while an immediately materialized draft handle is retained until the parent actually persists it. A regression proves discard still deletes that draft when the parent prop has not caught up.
- Fixed the three inline findings (duplicate schema field, CRD/domain layering, and late async materialization reopening a closed dialog) and resolved their threads.
- Declined the generic 50% docstring-coverage warning: this repository does not require docstrings on small internal callbacks, and adding narration to the local implementation would not improve the behavior contract. The exported draft hook already documents its lifecycle contract and the new edge cases are pinned by tests.

Evidence: signed fix commit `569792057`; latest `origin/develop` merged with signed commit `187daa31a`; concurrent remote feature merge reconciled without changing the gated tree in signed commit `6d3d3893e`. Node 24 gates on that exact tree: lint/typecheck/Biome/ESLint green, 354 test files / 2,999 passed / 2 skipped / 0 failed, production build green.

File: src/main/crdPages/space/callout/ResponseDefaultsConnector.tsx [hunk 4]
```diff
@@ -115,15 +133,70 @@ export function ResponseDefaultsConnector({
...
+        }
+      : undefined;
+
+  const cancelWhiteboardDraft = async () => {
+    dialogSessionRef.current += 1;
+    setWhiteboardEditorSession(undefined);
+    if (!whiteboardDraft?.handle || whiteboardDraft.handle.whiteboardID === initialDraftID.current) {
+      return true;
+    }
+    return whiteboardDraft.discard();
```
[the remark this answers is not in the document]
Comment (antst) [reply src/main/crdPages/space/callout/ResponseDefaultsConnector.tsx]: Fixed in 61c095af3. Cancellation now preserves only a present parent-committed handle whose whiteboard ID matches the opening draft; otherwise it calls discard, including the materialized-before-parent-commit window. The new regression reproduces that timing and the existing regression protects a previously committed draft.

**Comment by antst** [comment]
CodeRabbit incremental review triage completed on 61c095af3:

- The schema-sidebar out-of-scope warning is declined. Removing the duplicate InnovationFlowStateSettings.sidebar declaration was the necessary conflict repair after merging develop; retaining both declarations produces an invalid duplicate GraphQL field. No unrelated schema behavior was introduced.
- The generic docstring-coverage warning is declined. Repository convention avoids adding comments or docstrings to unchanged code, and the changed lifecycle behavior is covered by focused deterministic regressions. No missing contract was identified.
- The handleRef render-write nitpick was already addressed in signed commit 569792057: committed prop changes synchronize in an effect while the internal materialized handle remains available for cleanup. The new cancellation regression exercises that same boundary.

**Comment by antst** [comment]
@coderabbit review

**Review by polibb — APPROVED** [review]
